lib/couchbase/protostellar/cluster.rb

```diff
@@ -29,23 +29,49 @@
 module Couchbase
   module Protostellar
     class Cluster
+      VALID_CONNECTION_STRING_PARAMS = [
+        "trust_certificate",
+      ].freeze
+
       attr_reader :client
 
       def self.connect(connection_string, options = Couchbase::Options::Cluster.new)
-        connect_options = ConnectOptions.new(username: options.authenticator.username,
-                                             password: options.authenticator.password,
-                                             timeouts: Protostellar::Timeouts.from_cluster_options(options))
-        Cluster.new(connection_string.split("://")[1], connect_options)
+        params = parse_connection_string_params(connection_string)
+
+        connect_options = ConnectOptions.new(
+          username: options.authenticator.username,
+          password: options.authenticator.password,
+          timeouts: Protostellar::Timeouts.from_cluster_options(options),
+          root_certificates: params.key?("trust_certificate") ? File.read(params["trust_certificate"]) : nil
+        )
+        Cluster.new(connection_string.split("://")[1].split("?")[0], connect_options)
+      end
+
+      def self.parse_connection_string_params(connection_string)
+        params =
+          if connection_string.include? "?"
+            connection_string.split("?")[1].split("&").to_h { |p| p.split("=") }
+          else
+            {}
+          end
+
+        # Show warnings for the connection string parameters that are not supported
+        params.each do |k, v|
+          warn "Unknown parameter '#{k}' in connection string (value '#{v}')" unless VALID_CONNECTION_STRING_PARAMS.include?(k)
```

Comment:
I'm going to need a better solution for logging

Reply:
Standard library has `logger`, so maybe create one and pass it around in the options. So that the user will be able to override it. Or do like we are doing in Classic SDK, with `Couchbase.logger` (or even use it here) 

https://github.com/couchbase/couchbase-ruby-client/blob/4f3ac6381f10473ddf3d99f094a35e672458674e/lib/couchbase/logger.rb#L46